extensions/config-yaml/deployment/src/main/java/io/quarkus/config/yaml/deployment/ConfigYamlProcessor.java

```diff
@@ -17,10 +17,10 @@ public FeatureBuildItem feature() {
 
     @BuildStep
     public void bootstrap(
-            BuildProducer<AdditionalBootstrapConfigSourceProviderBuildItem> additionalBootstrapConfigSourceProvider) {
-        additionalBootstrapConfigSourceProvider.produce(new AdditionalBootstrapConfigSourceProviderBuildItem(
+            BuildProducer<AdditionalStaticInitConfigSourceProviderBuildItem> additionalBootstrapConfigSourceProvider) {
```

Comment:
We cannot replace the provider build item. It needs a new step to produce the `AdditionalStaticInitConfigSourceProviderBuildItem` and keep the `AdditionalBootstrapConfigSourceProviderBuildItem`.

I guess we don't have a test for the bootstrap part, so it would be great to add it as well to make sure it works on both phases.